scripts/sqllogictest/result.py

```diff
@@ -447,10 +447,11 @@ def matches_regex(input: str, actual_str: str) -> bool:
         should_match = False
         regex_str = input.replace("<!REGEX>:", "")
     # The exact match will never be the same, allow leading and trailing messages
-    if regex_str[:2] != '.*':
-        regex_str = ".*" + regex_str
-    if regex_str[-2:] != '.*':
-        regex_str = regex_str + '.*'
+    if should_match:
+        if regex_str[:2] != '.*':
```

Comment:
I don't think this is the correct way to fix this. This says, "If there should be a match, find the match as a substring or as the whole string".
This can also then mean, if there should not be a match, treat the regex string as is. This could also mean, the regex string must match the actual string completely. 

The regex string should be treated the same regardless of if there should be a match or not. I've tried to find a counter example, but in this exact scenario it's hard

We can talk about this after lunch

Reply:
Thank you for the detailed expansion!